docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+
+```
+/MyProject
+|__Program.cs
+|__Type1.cs
+|__Type2.cs
+|__project.json
+```
+
+This works great when the size of your project is relatively small.  However, if you have a larger app with many different data types and potentially multiple layers, you may wish to organize things logically.  This is where folders come into play.  You can either follow along with [the sample project](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/NewTypes) that this guide covers, or create your own files and folders.
+
+To begin, create a new folder under the root of your project.  `/Model` is chosen here.
+
+```
+/NewTypes
+|__/Model
+|__Program.cs
+|__project.json
+```
+
+Now add some new types to the folder:
+
+```
+/NewTypes
+|__/Model
+   |__Type1.cs
+   |__Type2.cs
+|__Program.cs
+|__project.json
+```
+
+Now, just as if they were files in the same directory, give them all the same namespace so you can include them in your `Program.cs`.
+
+### Example: Pet Types
+
+This example creates two new types, `Dog` and `Cat`, and has them implement an interface, `IPet`.
+
+Folder Structure:
+
+```
+/NewTypes
+|__/Types
+   |__Dog.cs
+   |__Cat.cs
+   |__IPet.cs
+|__Program.cs
+|__project.json
+```
+
+`IPet.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public interface IPet
+	{
+		string TalkToOwner();
+	}
+}
+```
+
+`Dog.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public class Dog : IPet
+	{
+		public string TalkToOwner() => "Woof!";
+	}
+}
+```
+
+`Cat.cs`:
+```csharp
+using System;
+
+namespace Pets
+{
+	public class Cat : IPet
+	{
+		public string TalkToOwner() => "Meow!";
+	}
+}
+```
+
+`Program.cs`:
+```csharp
+using System;
+using Pets;
+using System.Collections.Generic;
+
+namespace ConsoleApplication
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            List<IPet> pets = new List<IPet>
+            {
+                new Dog(),
+                new Cat()  
+            };
+            
+            foreach (var pet in pets)
+            {
+                Console.WriteLine(pet.TalkToOwner());
+            }
+        }
+    }
+}
+```
+
+`project.json`:
+```javascript
+{
+    "version": "1.0.0-*",
+    "compilationOptions": {
+        "emitEntryPoint": true
+    },
+
+    "dependencies": {
+        "Microsoft.NETCore.Runtime": "1.0.1-beta-*",
+        "System.IO": "4.0.11-beta-*",
+        "System.Console": "4.0.0-beta-*",
+        "System.Runtime": "4.0.21-beta-*",
+        "System.Collections": "4.0.11-beta-23516"
+    },
+
+    "frameworks": {
+        "dnxcore50": { }
+    }
+}
+```
+
+And if you run this:
+
+```
+$ dotnet restore
+$ dotnet run
+Woof!
+Meow!
+```
+
+New pet types can be added (such as a `Bird`), extending this project.
+
+## Testing your Console App
+
+You'll probably be wanting to test your projects at some point.  Here's a good way to do it:
```

Comment:
This is a good way, and I'm thinking that we should make this a proscribed way of doing things.